src/timetable_gui.cpp

```diff
@@ -490,19 +533,28 @@ struct TimetableWindow : Window {
 		}
 		tr.top += GetCharacterHeight(FS_NORMAL);
 
+		/* Draw the lateness display, or indicate that the timetable has not started yet. */
 		if (v->timetable_start != 0) {
 			/* We are running towards the first station so we can start the
 			 * timetable at the given time. */
-			SetDParam(0, STR_JUST_DATE_TINY);
-			SetDParam(1, GetDateFromStartTick(v->timetable_start));
-			DrawString(tr, STR_TIMETABLE_STATUS_START_AT);
+			if (_settings_client.gui.timetable_mode == TimetableMode::Seconds) {
+				/* Real time units use seconds relative to now. */
+				SetDParam(0, (static_cast<TimerGameTick::Ticks>(v->timetable_start - TimerGameTick::counter) / Ticks::TICKS_PER_SECOND));
+				DrawString(tr, STR_TIMETABLE_STATUS_START_IN_SECONDS);
+			} else {
+				/* Calendar units use dates. */
+				SetDParam(0, STR_JUST_DATE_TINY);
+				SetDParam(1, GetDateFromStartTick(v->timetable_start));
+				DrawString(tr, STR_TIMETABLE_STATUS_START_AT_DATE);
+			}
 		} else if (!HasBit(v->vehicle_flags, VF_TIMETABLE_STARTED)) {
-			/* We aren't running on a timetable yet, so how can we be "on time"
-			 * when we aren't even "on service"/"on duty"? */
+			/* We aren't running on a timetable yet. */
 			DrawString(tr, STR_TIMETABLE_STATUS_NOT_STARTED);
-		} else if (v->lateness_counter == 0 || (!_settings_client.gui.timetable_in_ticks && v->lateness_counter / Ticks::DAY_TICKS == 0)) {
+		} else if (v->lateness_counter == 0 || (_settings_client.gui.timetable_mode != TimetableMode::Ticks && v->lateness_counter / TicksPerDayOrSecond() == 0)) {
```

Comment:
Special-case that could be removed if `TicksPerDayOrSecond()` also handled `TimetableMode::Ticks`.